sql-plugins/src/test/java/io/cdap/pipeline/sql/plugins/aggregator/GroupByAggregatorTest.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.pipeline.sql.plugins.aggregator;
+
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rex.RexInputRef;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.tools.RelBuilder;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GroupByAggregatorTest {
+  @Test
+  public void testNullAggregates() {
+    GroupByConfig config = new GroupByConfig("a", null);
+    Assert.assertEquals(0, config.getAggregates().size());
+  }
+
+  @Test
+  public void testEmptyAggregates() {
+    GroupByConfig config = new GroupByConfig("a", "");
+    Assert.assertEquals(0, config.getAggregates().size());
+  }
+
+  @Test
+  public void testNullGroupByFields() {
+    GroupByConfig config = new GroupByConfig(null, "a:COUNT(b)");
+    Assert.assertEquals(0, config.getGroupByFields().size());
+  }
+
+  @Test
+  public void testEmptyGroupByFields() {
+    GroupByConfig config = new GroupByConfig("", "a:COUNT(b)");
+    Assert.assertEquals(0, config.getGroupByFields().size());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testInvalidGroupByField() {
+    GroupByConfig config = new GroupByConfig(",", "a:COUNT(b)");
```

Comment:
it looks like this can be tested by calling methods on the config object directly rather than having the extra layer of going through the aggregator.

Reply:
same for several of the methods below

Reply:
Fixed